platform-core/src/main/kotlin/org/dashevo/platform/Names.kt

```diff
@@ -0,0 +1,294 @@
+/**
+ * Copyright (c) 2020-present, Dash Core Group
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+package org.dashevo.platform
+
+import org.bitcoinj.core.Base58
+import org.bitcoinj.core.ECKey
+import org.bitcoinj.core.Sha256Hash
+import org.bitcoinj.params.EvoNetParams
+import org.dashevo.dapiclient.DapiClient
+import org.dashevo.dapiclient.model.DocumentQuery
+import org.dashevo.dpp.document.Document
+import org.dashevo.dpp.identity.Identity
+import org.dashevo.dpp.util.Entropy
+import org.json.JSONObject
+import java.io.ByteArrayOutputStream
+import java.lang.Thread.sleep
+
+class Names (val platform: Platform) {
+
+    companion object {
+        const val DEFAULT_PARENT_DOMAIN = "dash"
+    }
+
+    fun register(name: String, identity: Identity, identityHDPrivateKey: ECKey): Document
+    {
+        val dpp = platform.dpp
+
+        val identityType = if (identity.type.value == 2) "application" else "user"
+
+        // @ts-ignore
+
+        val records = HashMap<String, Any?>(1)
+        records["dashIdentity"] = identity.id
+
+        val nameSlice = name.indexOf('.')
+        val normalizedParentDomainName =
+            if(nameSlice == -1) "dash" else name.slice(nameSlice + 1 .. name.length)
+
+        val label = if(nameSlice == -1) name else name.slice(0 .. nameSlice)
+
+        val normalizedLabel = label.toLowerCase();
+        val fullDomainName = "$normalizedLabel.$normalizedParentDomainName";
+
+        val nameHash = Sha256Hash.twiceOf(fullDomainName.toByteArray())
+        val nameHashHex = nameHash.toString()
+
+        val preorderSaltBase58 = Entropy.generate();
+        val preOrderSaltRaw = Base58.decode(preorderSaltBase58)
+
+        val baos = ByteArrayOutputStream(preOrderSaltRaw.size + nameHash.bytes.size)
+        baos.write(preOrderSaltRaw)
+        baos.write(0x56)
```

Comment:
What the meaning of these values? Can we store them in constants or add an inline comment?

Reply:
"5620" is a prefix used to identity a string as an SHA256D hash.  56 for 256 and 20 (hex) for 32 bytes.  I will do constants in the code.

Reply:
This commit addresses this issue:
https://github.com/dashevo/android-dashpay/pull/1/commits/d96ee2b68d847bc4d4d2458676ef4d2e375e2556